### DISCUSSION TITLE
*Ts hint as string literal

### DISCUSSION THREAD

**Discussion by pinterior** [microsoft/pyright#4256] [General]
string-literal type hints starts with `*` like `x: Tuple["*Ts"]` or `*args: "*Ts"` cause following error:

```
error: Unpack operation not allowed in this context
```
is this the desired behavior?

**Comment by erictraut** [comment]
Why are you using a stringified type annotation (a forward declaration) in this case? Is there a compelling need to do this? Type arguments shouldn't be forward declared. If you need to enclose the surrounding type in a forward declaration (e.g. `"MyClass[*Ts]"`), that should work.

**Reply by pinterior** [reply]
to suppress SyntaxError from Python 3.8. (and it's easier to read than Unpack)

**Reply by erictraut** [reply]
You'll need to use `Unpack` for this. The problem is that the quoted section needs to be a syntactically valid expression. The text `*Ts` is not a valid expression because `*` is not an operator in this context.

**Reply by pinterior** [reply]
i understood literal has a different context than bare annotation.
alright, thank you.
